### ISSUE TITLE
"Highlight all matches" in find bar does not work

### ISSUE THREAD

**Issue by worblehat** [qterminal/qterminal#120]
Enabling "Highlight all matches" in the find bar options does not make any difference. Only the current match is highlighted.

**Comment by surlykke** [comment]
2015-02-23 9:51 GMT+01:00 Tobias Marquardt <notifications@github.com>:

> Enabling "Highlight all matches" in the find bar options does not make any
> difference. Only the current match is highlighted.
>
>
Hmm. It's been a while since I coded this. Not sure the 'Highlight all..'
ever worked. Maybe it only made it to the todo list :-). Anyways, I'll look
into it.

br. Chr.

**Comment by jleclanche** [comment]
@surlykke tbh I think highlight all matches should be default behaviour.

**Comment by jleclanche** [comment]
@surlykke ping

**Comment by f2404** [comment]
No, it doesn't seem it has ever worked - the checkbox handler is missing from `QTermWidget`.

**Comment by surlykke** [comment]
2016-05-31 18:14 GMT+02:00 Igor <notifications@github.com>:

> No, it doesn't seem it has ever worked - the checkbox handler is missing
> from QTermWidget.
>
> No it has never worked. I'm responsible for putting the checkbox there. It
was my intention to implement show-all-matches functionality, but I never
got around to it.

br. Chr.

**Comment by yacoob** [comment]
Any chance of implementing that any time soon? I'm looking for terminal emulators that have a function like that (highlighting a phrase through the screen) - and qterminal is one of the few that gets close enough :)

**Comment by tsujan** [comment]
> can confirm still the checkbox has no effect

Apparently it was never implemented. In a text editor (`QPlainTextEdit`) highlighting and selecting have totally different meanings: you can have only one continuous selection but various kinds of highlighted texts at the same time.

It doesn't seem easy to me, although I haven't looked into it yet.
